src/cloudwatch-mcp-server/awslabs/cloudwatch_mcp_server/cloudwatch_alarms/tools.py

```diff
@@ -0,0 +1,656 @@
+            alarm history with time range suggestions, or component alarm details for composite alarms.
+
+        Example:
+            result = await get_alarm_history(
+                ctx,
+                alarm_name="my-cpu-alarm",
+                start_time="2025-06-18T00:00:00Z",
+                end_time="2025-06-19T00:00:00Z"
+            )
+            if isinstance(result, AlarmHistoryResponse):
+                print(f"Found {len(result.history_items)} history items")
+                for suggestion in result.time_range_suggestions:
+                    print(f"Suggested investigation time range: {suggestion.start_time} to {suggestion.end_time}")
+        """
+        try:
+            # Use default client
+            cloudwatch_client = self.cloudwatch_client
+            
+            # Set up default time range (last 24 hours)
+            if end_time is None or not isinstance(end_time, str):
+                end_time = datetime.now()
+            else:
+                end_time = datetime.fromisoformat(end_time.replace('Z', '+00:00'))
+            
+            if start_time is None or not isinstance(start_time, str):
+                start_time = end_time - timedelta(days=1)
+            else:
+                start_time = datetime.fromisoformat(start_time.replace('Z', '+00:00'))
+            
+            # Set default history item type
+            if not history_item_type:
+                history_item_type = 'StateUpdate'
+            
+            logger.info(f'Fetching alarm history for {alarm_name}')
+            logger.info(f'Time range: {start_time} to {end_time}')
+            
+            # Make the basic API call to GetAlarmHistory
+            params = {
+                'AlarmName': alarm_name,
+                'StartDate': start_time,
+                'EndDate': end_time,
+                'HistoryItemType': history_item_type,
+                'MaxRecords': max_items
+            }
+            
+            response = cloudwatch_client.describe_alarm_history(**params)
+            
+            # Transform history items with enhanced parsing
+            history_items = []
+            for item in response.get('AlarmHistoryItems', []):
+                history_item = self._transform_history_item(item)
+                history_items.append(history_item)
+            
+            # Get detailed alarm information
+            alarm_details = await self._get_alarm_details(cloudwatch_client, alarm_name)
+            
+            # Handle composite alarms if requested
+            if include_component_alarms and alarm_details.alarm_type == "CompositeAlarm":
+                return await self._handle_composite_alarm(cloudwatch_client, alarm_details)
+            
+            # Generate time range suggestions based on alarm history and configuration
+            time_range_suggestions = self._generate_time_range_suggestions(history_items, alarm_details)
+            
+            # Handle pagination
```

Comment:
nit: we're not really handling pagination here, just detecting that we have more results.